src/provisioner/cluster.py

```diff
@@ -65,17 +78,106 @@ def _get_kvs(self, prefix, node):
         elif isinstance(node, list):
             for i, n in enumerate(node):
                 kvs.extend(self._get_kvs(f'{prefix}[{i}]', n))
-        elif isinstance(node, str):
+        elif isinstance(node, str) or isinstance(node, int):
             kvs.append((prefix, node))
         return kvs
 
     def save(self, config_store):
         """ Saves cluster information onto the conf store """
 
         kvs = []
-        for node in self._node_list:
-            node_id = node['id']
-            key_prefix = f'node>{node_id}'
-            kvs.extend(self._get_kvs(key_prefix, node))
+        try:
+            for node in self._node_list:
+                node_id = node.pop('id')
+                components = {}
+                for component in node.pop('components'):
```

Comment:
This code needs to be in CortxProvisioner::config_apply_cluster(cortx_config_store) function because we are doing similar processing there.

Reply:
Done.